st2api/tests/unit/controllers/v1/test_executions_simple.py

```diff
@@ -233,6 +234,22 @@ def test_post_delete(self):
         self.assertEqual(delete_resp.status_int, 200)
         self.assertEqual(delete_resp.json['status'], 'canceled')
 
+    def test_post_delete_trace(self):
+        LIVE_ACTION_TRACE = copy.copy(LIVE_ACTION_1)
+        LIVE_ACTION_TRACE['context'] = {'trace_context': {'trace_tag': 'balleilaka'}}
+        post_resp = self._do_post(LIVE_ACTION_TRACE)
+        self.assertEqual(post_resp.status_int, 201)
+        delete_resp = self._do_delete(self._get_actionexecution_id(post_resp))
+        self.assertEqual(delete_resp.status_int, 200)
+        self.assertEqual(delete_resp.json['status'], 'canceled')
+        trace_id = str(Trace.get_all()[0].id)
+        LIVE_ACTION_TRACE['context'] = {'trace_context': {'id_': trace_id}}
+        post_resp = self._do_post(LIVE_ACTION_TRACE)
+        self.assertEqual(post_resp.status_int, 201)
+        delete_resp = self._do_delete(self._get_actionexecution_id(post_resp))
+        self.assertEqual(delete_resp.status_int, 200)
+        self.assertEqual(delete_resp.json['status'], 'canceled')
```

Comment:
Should you be checking if the trace eventually contains both executions? I am not exactly sure what this test achieves otherwise.